packages/contracts/contracts/voting/majority/MajorityVotingBase.sol

```diff
@@ -297,15 +297,59 @@ abstract contract MajorityVotingBase is
             revert VoteParticipationExceeded({limit: PCT_BASE, actual: _participationRequiredPct});
         }
 
-        if (_minDuration == 0) {
-            revert VoteDurationZero();
+        if (_minDuration < 60 minutes) {
+            revert MinDurationOutOfBounds({limit: 60 minutes, actual: _minDuration});
+        }
+
+        if (_minDuration > 365 days) {
+            revert MinDurationOutOfBounds({limit: 365 days, actual: _minDuration});
         }
 
         participationRequiredPct = _participationRequiredPct;
         supportRequiredPct = _supportRequiredPct;
         minDuration = _minDuration;
     }
 
+    /// @notice Validates and returns the proposal vote dates.
+    /// @param _start The start date of the proposal vote. If 0, the current timestamp is used and the vote starts immediately.
+    /// @param _end The end date of the proposal vote. If 0, `_start + minDuration` is used.
+    /// @return startDate The validated start date of the proposal vote.
+    /// @return endDate The validated end date of the proposal vote.
+    function _validateVoteDates(uint64 _start, uint64 _end)
+        internal
+        view
+        returns (uint64 startDate, uint64 endDate)
+    {
+        uint64 currentTimestamp = getTimestamp64();
+
+        if (_start == 0) {
+            startDate = currentTimestamp;
+        } else {
+            startDate = _start;
+
+            if (startDate < currentTimestamp) {
+                revert DateOutOfBounds({limit: currentTimestamp, actual: startDate});
+            }
+        }
+        // Check if `startDate` is after the latest possible start date in which case the addition of `minDuration` would lead to an integer overflow.
+        uint64 latestStartDate = type(uint64).max - minDuration;
```

Comment:
is this check really necessary? the latest possible date (uint64 max) is so far in the future that humanity pobably has already died out

Reply:
This check addresses HAL-03:
> The second case is missing validation of the `_startDate` parameter. When
its value is close to the uint64 maximum, the transaction might revert due
to integer overflow caused by the addition of `_startDate` and `minDuration`